### PR TITLE
inventoryMove and shipment changes

### PR DESCRIPTION
**Pull request by LakhwinderVienna**
ShipmentnMoveConfirmation Changes

### CODE REVIEW

File: ModelLibrary/Model/MMovement.cs [hunk 1]
```diff
@@ -360,6 +370,16 @@ public String PrepareIt()
                 }
             }
 
+
+            //Lakhwinder 10 Feb 2021
+            string s = CheckConfimationDocType();
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovement.cs:375]: Use this condition with InTransit Document Type

File: ModelLibrary/Model/MInOutConfirm.cs [hunk 2]
```diff
@@ -1144,8 +1150,14 @@ private bool CreateDifferenceDoc(MInOut inout, MInOutLineConfirm confirm)
                     _inventory = new MInventory(wh);
                     _inventory.SetDescription(Msg.Translate(GetCtx(),
                         "M_InOutConfirm_ID") + " " + GetDocumentNo());
+
+                    //Lakhwinder
+                    int doctypeSetting = CheckAssociateDocTypeSetting(inout.GetC_DocType_ID(), false);
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MInOutConfirm.cs:1155]: Check Columns on Document type if exist or not

File: ModelLibrary/Model/MInOut.cs [hunk 3]
```diff
@@ -2152,6 +2161,18 @@ public virtual String CompleteIt()
             }
 
 
+            //Lakhwinder 10 Feb 2021
+            if (MDocType.Get(GetCtx(), GetC_DocType_ID()).IsShipConfirm())
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MInOut.cs:2165]: Use single object to get multiple values

File: ModelLibrary/Model/MMovementConfirm.cs [hunk 4]
```diff
@@ -83,6 +89,14 @@ public MMovementConfirm(MMovement move)
         {
             SetClientOrg(move);
             SetM_Movement_ID(move.GetM_Movement_ID());
+
+            //Lakhwinder 1Feb2021
+            //Shipment and Inventory Move Module Changes
+            //Solution Proposed Puneed 
+            //STandard changes Analysed in Gull Implementation
+            //SetC_DocType_ID( move.GetC_DocType_ID());
+            SetC_DocType_ID(Util.GetValueOfInt(MDocType.Get(GetCtx(), move.GetC_DocType_ID()).Get_Value("C_DocTypeConfrimation_ID")));
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovementConfirm.cs:98]: check column if exist or not

File: ModelLibrary/Model/MInOutConfirm.cs [hunk 5]
```diff
@@ -332,6 +332,12 @@ public MInOutConfirm(MInOut ship, String confirmType)
             SetClientOrg(ship);
             SetM_InOut_ID(ship.GetM_InOut_ID());
             SetConfirmType(confirmType);
+
+            //Lakhwinder 1Feb2021
+            //Shipment and Inventory Move Module Changes
+            //Solution Proposed Puneed 
+            //STandard changes Analysed in Gull Implementation
+            SetC_DocType_ID(Util.GetValueOfInt(MDocType.Get(GetCtx(), ship.GetC_DocType_ID()).Get_Value("C_DocTypeConfrimation_ID")));
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MInOutConfirm.cs:340]: check column if exist or not

File: ModelLibrary/Model/MMovementConfirm.cs [hunk 6]
```diff
@@ -115,6 +129,13 @@ public static MMovementConfirm Create(MMovement move, Boolean checkExisting)
                 // setting QtyEntered in Target Qty on Confirmation Line
                 cLine.SetTargetQty(mLine.GetQtyEntered());
                 cLine.SetConfirmedQty(mLine.GetQtyEntered());
+
+                //Lakhwinder 1Feb2021
+                //Shipment and Inventory Move Module Changes
+                //Solution Proposed Puneed 
+                //STandard changes Analysed in Gull Implementation
+                cLine.SetC_UOM_ID(mLine.GetC_UOM_ID());
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovementConfirm.cs:137]: check column if exist or not

File: ModelLibrary/Model/MMovementConfirm.cs [hunk 7]
```diff
@@ -776,115 +804,276 @@ private Boolean CreateDifferenceDoc(MMovement move, MMovementLineConfirm confirm
...
-                //	Get Warehouse for TarGet
-                MLocator loc = MLocator.Get(GetCtx(), mLine.GetM_LocatorTo_ID());
-                if (_inventoryTo != null
-                    && _inventoryTo.GetM_Warehouse_ID() != loc.GetM_Warehouse_ID())
-                    _inventoryTo = null;
 
-                if (_inventoryTo == null)
+                //Lakhwinder
+                //15Feb2021 Create InventoryMove as Per setting on DoctType
+                int inventorySetting = CheckAssociateDocTypeSetting(move.GetC_DocType_ID(), false);
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovementConfirm.cs:925]: check column if exist or not

File: ModelLibrary/Model/MMovementConfirm.cs [hunk 8]
```diff
@@ -776,115 +804,276 @@ private Boolean CreateDifferenceDoc(MMovement move, MMovementLineConfirm confirm
...
+                        momementDiffrence.SetMovementDate(move.GetMovementDate());
+                        momementDiffrence.SetDescription(move.GetDescription() + "<-->" + move.GetDocumentNo());
+                        momementDiffrence.SetDTD001_MWarehouseSource_ID(move.GetDTD001_MWarehouseSource_ID());
+                        momementDiffrence.SetM_Warehouse_ID(move.GetM_Warehouse_ID());
+                        //movementScrap.SetC_IncoTerm_ID(GetC_IncoTerm_ID());
+                        momementDiffrence.SetC_Campaign_ID(move.GetC_Campaign_ID());
+                        momementDiffrence.SetDTD001_Packing_ID(move.GetDTD001_Packing_ID());
+                        momementDiffrence.SetDocStatus("DR");
+                        if (!momementDiffrence.Save(Get_Trx()))
+                        { log.Warning("CreateMaterialMoveDoc"); }
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovementConfirm.cs:828]: if header not saved, then no need to create lines.

File: ModelLibrary/Model/MMovementConfirm.cs [hunk 9]
```diff
@@ -776,115 +804,276 @@ private Boolean CreateDifferenceDoc(MMovement move, MMovementLineConfirm confirm
...
+                        movementScrap.SetMovementDate(move.GetMovementDate());
+                        movementScrap.SetDescription(move.GetDescription() + "<-->" + move.GetDocumentNo());
+                        movementScrap.SetDTD001_MWarehouseSource_ID(move.GetDTD001_MWarehouseSource_ID());
+                        movementScrap.SetM_Warehouse_ID(move.GetM_Warehouse_ID());
+                        //movementScrap.SetC_IncoTerm_ID(GetC_IncoTerm_ID());
+                        movementScrap.SetC_Campaign_ID(move.GetC_Campaign_ID());
+                        movementScrap.SetDTD001_Packing_ID(move.GetDTD001_Packing_ID());
+                        movementScrap.SetDocStatus("DR");
+                        if (!movementScrap.Save(Get_Trx()))
+                        { log.Warning("CreateMaterialMoveDoc"); }
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovementConfirm.cs:943]: if header not saved then no need to save the lines.

File: ModelLibrary/Model/MMovementLineConfirm.cs [hunk 10]
```diff
@@ -108,10 +108,23 @@ public Boolean ProcessLine()
         {
             MMovementLine line = GetLine();
 
-            line.SetTargetQty(GetTargetQty());
-            line.SetMovementQty(GetConfirmedQty());
-            line.SetConfirmedQty(GetConfirmedQty());
-            line.SetScrappedQty(GetScrappedQty());
+
+            MProduct _Pro = new MProduct(GetCtx(),line.GetM_Product_ID(), Get_TrxName());
+            if (GetC_UOM_ID() != _Pro.GetC_UOM_ID())
```
Comment (Bharat-Vienna) [inline ModelLibrary/Model/MMovementLineConfirm.cs:113]: check UOM column if exist or not

**Review by karanvienna — COMMENTED** [review]
MSetup class is Okay

**Review by Bharat-Vienna — APPROVED** [review]
Fine

**Pull request merged**